### ISSUE TITLE
Failed to load resource: the server responded with a status of 500 ()

### ISSUE THREAD

**Issue by the-bort-the** [9p4/jellyfin-plugin-sso#223]
**Describe the bug**
After enabling SSO for jellyfin using OAuth2/OpenID Authentik provider, I'm seeing the below errors in the browser and jellyfin logs.
https://docs.goauthentik.io/integrations/services/jellyfin/

"Error processing request." (seen in browser after clicking SSO button)

Also seen after opening dev tools on Brave and Firefox:
"Failed to load resource: the server responded with a status of 500 ()" 
![jellyfin_error](https://github.com/user-attachments/assets/5ab2d7f4-dcc4-4829-9290-02d9ce7c43a5)

from jellyfin logs:
[16:26:32] [INF] [36] Jellyfin.Api.Auth.CustomAuthenticationHandler: CustomAuthentication was not authenticated. Failure message: Invalid token.
[16:26:32] [INF] [36] Jellyfin.Api.Auth.CustomAuthenticationHandler: CustomAuthentication was not authenticated. Failure message: Invalid token.
[16:26:32] [INF] [36] Jellyfin.Api.Auth.CustomAuthenticationHandler: AuthenticationScheme: CustomAuthentication was challenged.
[16:26:51] [INF] [56] Jellyfin.Plugin.SSO_Auth.Api.SSOController: SSO Controller initialized
[16:26:51] [ERR] [56] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL GET /sso/OID/start/authentik.
System.InvalidOperationException: Error loading discovery document: Error connecting to https://jellyfin.example.com/application/o/jellyfin/.well-known/openid-configuration: Not Found

![jellyfin_authentik_1](https://github.com/user-attachments/assets/27f89957-8348-4a1f-9272-c8c1f0243e4b)
![jellyfin_branding](https://github.com/user-attachments/assets/ce6428d2-cb21-496b-93ca-d4df4fdb3aab)
![jellyfin_plugin_settings_2](https://github.com/user-attachments/assets/0f6b395e-e1c8-4722-bbe5-ee3b0c48fed0)
![jellyfin_plugin_settings_1](https://github.com/user-attachments/assets/9a36cae3-8541-4cc6-b190-2131729b8232)
![jellyfin_plugin](https://github.com/user-attachments/assets/42d54ac0-055b-4f75-acf4-ec5c2a939661)
![jellyfin_repos](https://github.com/user-attachments/assets/02d458f6-b0c1-4869-b335-fa6afcdfdda5)

**To Reproduce**
After following the steps on jellyfin and authentik docs to implement, I'm seeing this error. I will say it was working just fine for about a week until now. By the time I discovered the error, I also noticed the plugin was not installed anymore. I had to install the plugin like I did the first time.

**Expected behavior**
SSO button logs me in with the authentik user

**Screenshots**
added screenshots of config within authentik and jellyfin

**Configuration**
where is the xml stored? I'm running this inside a container

**Versions (please complete the following information):**

- OS: Debian
- Browser: Brave, Chrome, Firefox
- Jellyfin Version: `jellyfin/jellyfin:10.9.11-amd64.20240907-221241` (worked originally until error above) also tried on this newer tag, but ran into the same error: `jellyfin/jellyfin:2024120205-amd64`
- Plugin Version: 3.5.2.4

**Additional context**
- Was working for about a week
- QuickConnect is enabled
- Should also note, other applications, providers are working within Authentik and I'm able to still access Jellyfin via already configured user/password

**Comment by 9p4** [comment]
Can you remove the well-known bit from your endpoint in the plugin config?

**Comment by the-bort-the** [comment]
Removed it and still same thing. I am noticing a lot of this from the logs. Additionally, when I restart after saving, the plugin isn't loading the saved provider.

```
[00:42:33] [INF] [26] Jellyfin.Api.Auth.CustomAuthenticationHandler: CustomAuthentication was not authenticated. Failure message: Invalid token.
[00:42:33] [INF] [26] Jellyfin.Api.Auth.CustomAuthenticationHandler: CustomAuthentication was not authenticated. Failure message: Invalid token.
[00:42:33] [INF] [26] Jellyfin.Api.Auth.CustomAuthenticationHandler: AuthenticationScheme: CustomAuthentication was challenged.
```

Here is the xml also:
```
<?xml version="1.0" encoding="utf-8"?>
<PluginConfiguration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <SamlConfigs />
  <OidConfigs>
    <item>
      <key>
        <string />
      </key>
      <value>
        <PluginConfiguration>
          <Enabled>false</Enabled>
          <EnableAuthorization>false</EnableAuthorization>
          <EnableAllFolders>false</EnableAllFolders>
          <EnabledFolders />
          <AdminRoles />
          <Roles />
          <EnableFolderRoles>false</EnableFolderRoles>
          <EnableLiveTvRoles>false</EnableLiveTvRoles>
          <EnableLiveTv>false</EnableLiveTv>
          <EnableLiveTvManagement>false</EnableLiveTvManagement>
          <LiveTvRoles />
          <LiveTvManagementRoles />
          <FolderRoleMappings />
          <OidScopes />
          <NewPath>false</NewPath>
          <CanonicalLinks />
          <DisableHttps>false</DisableHttps>
          <DoNotValidateEndpoints>false</DoNotValidateEndpoints>
          <DoNotValidateIssuerName>false</DoNotValidateIssuerName>
        </PluginConfiguration>
      </value>
    </item>
    <item>
      <key>
        <string>authentik</string>
      </key>
      <value>
        <PluginConfiguration>
          <OidEndpoint>https://authentik.example.com/application/o/jellyfin/openid-configuration</OidEndpoint>
          <OidClientId>redacted</OidClientId>
          <OidSecret>redacted</OidSecret>
          <Enabled>true</Enabled>
          <EnableAuthorization>true</EnableAuthorization>
          <EnableAllFolders>true</EnableAllFolders>
          <EnabledFolders>
            <string>f78d974185fec8586ec7f643d6e2151f</string>
          </EnabledFolders>
          <AdminRoles />
          <Roles />
          <EnableFolderRoles>false</EnableFolderRoles>
          <EnableLiveTvRoles>false</EnableLiveTvRoles>
          <EnableLiveTv>false</EnableLiveTv>
          <EnableLiveTvManagement>false</EnableLiveTvManagement>
          <LiveTvRoles />
          <LiveTvManagementRoles />
          <FolderRoleMappings />
          <OidScopes />
          <NewPath>true</NewPath>
          <CanonicalLinks />
          <DisableHttps>false</DisableHttps>
          <DoNotValidateEndpoints>false</DoNotValidateEndpoints>
          <DoNotValidateIssuerName>false</DoNotValidateIssuerName>
        </PluginConfiguration>
      </value>
    </item>
  </OidConfigs>
</PluginConfiguration>
```

![jellyfin_not_loading](https://github.com/user-attachments/assets/387b4d7f-c741-4c1e-b2d3-cfe3e4493f19)

**Comment by 9p4** [comment]
Can you post the entire log file (with secrets redacted)?

**Comment by the-bort-the** [comment]
[jellyfin_logs.txt](https://github.com/user-attachments/files/18216796/jellyfin_logs.txt)

**Comment by 9p4** [comment]
If you go to `https://authentik.example.com/application/o/jellyfin/openid-configuration/.well-known/openid-configuration` in your browser, what does it return?

**Comment by the-bort-the** [comment]
That results in 404's from the browser. I still have the configuration without `.well-known` in the plugin config. This reflects in the logs also/

![jellyfin_authentik_not_found](https://github.com/user-attachments/assets/ff5a5cf2-6a8a-48d7-9b8c-50648e5139cb)

```
[18:27:44] [INF] [150] Jellyfin.Plugin.SSO_Auth.Api.SSOController: SSO Controller initialized
[18:27:44] [ERR] [150] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL GET /sso/OID/start/authentik.
System.InvalidOperationException: Error loading discovery document: Error connecting to https://authentik.example.com/application/o/jellyfin/openid-configuration/.well-known/openid-configuration: Not Found
   at IdentityModel.OidcClient.OidcClient.EnsureProviderInformationAsync(CancellationToken cancellationToken) in /_/src/OidcClient/OidcClient.cs:line 413
   at IdentityModel.OidcClient.OidcClient.EnsureConfigurationAsync(CancellationToken cancellationToken) in /_/src/OidcClient/OidcClient.cs:line 374
   at IdentityModel.OidcClient.OidcClient.PrepareLoginAsync(Parameters frontChannelParameters, CancellationToken cancellationToken) in /_/src/OidcClient/OidcClient.cs:line 111
   at Jellyfin.Plugin.SSO_Auth.Api.SSOController.OidChallenge(String provider, Boolean isLinking)
   at lambda_method718(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

**Comment by the-bort-the** [comment]
So I believe I have figured things out. During the deployment of Authentik, I was also setting Caddy reverse proxy. Each application was being added to caddy in their respective compose.yml file. `caddy_network` was created as an external network, and all apps we added to it. 

This created a change for the jellyfin compose.yml in that I switched from `network_mode: host` to that of the `caddy_network` and one for jellyfin, `jellyfin_network`. This still wasn't enough. 

I added the `authentik_network` to jellyfin's compose.yml and now SSO is working. I also added the `.well-known` back to the endpoint. Some glue or cache must have held on for a week, before things broke. Not sure how to explain that. What's interesting is none of the other applications, seem to also need the `authentik_network` added to their compose.yml configurations; they only contain a network of their service name and the `caddy_network`.
